File: src/com/dafttech/terra/engine/gui/anchors/GUIAnchorSet.java
package com.dafttech.terra.engine.gui.anchors;

import java.util.ArrayList;
import java.util.List;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public class GUIAnchorSet {
    List<GUIAnchor> anchors = new ArrayList<GUIAnchor>();

    public void applyAnchorSet(GUIObject object, GUIContainer container) {
        for (GUIAnchor a : anchors) {
            a.applyAnchor(object, container);
        }
    }

    public boolean needsApplyOnFrame() {
        for (GUIAnchor a : anchors) {
            if (a.needsApplyOnFrame()) return true;
        }
        return false;
    }

    public boolean isContainerDependent() {
        for (GUIAnchor a : anchors) {
            if (a.isContainerDependent()) return true;
        }
        return false;
    }

    public void addAnchor(GUIAnchor anchor) {
        anchors.add(anchor);
    }
}

File: terra/src/main/scala/com/dafttech/terra/engine/gui/anchors/GUIAnchorSet.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

import scala.collection.mutable.ArrayBuffer

class GUIAnchorSet {
  private val anchors: ArrayBuffer[GUIAnchor] = new ArrayBuffer[GUIAnchor]

  def this(can: GUIAnchor*) {
    this()
    anchors ++= can
  }

  def applyAnchorSet(gObj: GUIObject, container: GUIContainer): Unit = {
    for (a <- anchors) {
      a.applyAnchor(gObj, container)
    }
  }

  def needsApplyOnFrame: Boolean = {
    for (a <- anchors) {
      if (a.needsApplyOnFrame) return true
    }
    false
  }

  def isContainerDependent: Boolean = {
    for (a <- anchors) {
      if (a.isContainerDependent) return true
    }
    false
  }

  def addAnchor(anchor: GUIAnchor): GUIAnchorSet = {
    anchors += anchor
    this
  }
}